src/imap.rs

```diff
@@ -1411,19 +1418,27 @@ impl Imap {
                         err
                     );
                 }
-            };
-
-            while let Some(Ok(msg)) = msgs.next().await {
```

Comment:
Looks like there was a bug -- we just stopped processing messages if stumbled upon a failed one, and went to the next intem in `sets`. Replaced this with skipping of failed-to-fetch messages

Reply:
Ok, added your comment to the code where i drain remaining messages